### DISCUSSION TITLE
Use semver for end-user version display

### DISCUSSION THREAD

**Discussion by oplik0** [solokeys/solo2-cli#29] [General]
CalVer is not something most people expect, it doesn't match the GitHub releases and it doesn't actually work as it returns 1:20200101.8 for 1.0.8 released on 2021-11-11 (which is the date 0 in lpc55 calver from my quick check).

See this discussion for an example of confusion: https://github.com/solokeys/kickstarter2021/discussions/31

It's not clear at all that this version is actually 1.0.8 from GitHub.

lpc55-host also implements [`to_semver`](https://github.com/lpc55/lpc55-host/blob/6ae35e1ffcd459c92fc74ad335ee330f65997d58/src/secure_binary.rs#L926-L928), so it should be a simple replacement.

**Comment by nickray** [comment]
It's unfortunate that the existing releases used semver, not calver, as planned.

The rationale for using calver is that semver kinda doesn't make sense bundled firmware releases (which in time might not be "one thing", but a tested compositum of a selection of apps, current for a given build date).

The owner really only cares:
- is this update going to break my key (i.e. requires a reset losing the secrets)? That is what the major release signifies, and we'll try to keep these as infrequent as possible.
- how recent is my firmware (i.e. should I updated?) and what changed since the one I have? The second question is answered by checking the build date against what was then the "latest stable" release of each included app.

Turning this into a discussion, in case the calver plan is controversial.
